lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscoverySpec.hs

```diff
@@ -312,6 +332,77 @@ prop_lookupDiscovered (s0, addr) =
             expectationFailure "couldn't find private key corresponding to addr"
 
 
+{-------------------------------------------------------------------------------
+                        Properties for CompareDiscovery
+-------------------------------------------------------------------------------}
+
+prop_compareKnownUnknown
+    :: (SeqState DummyTarget, ShowFmt Address, ShowFmt Address)
+    -> Property
+prop_compareKnownUnknown (s, ShowFmt known, ShowFmt addr) =
+    fst (isOurs known s) ==> case isOurs addr s of
+        (False,_) -> cover 5 True "known-unknown" $ prop LT
+        (True, _) -> property True
+  where
+    prop ordering = compareDiscovery s known addr === ordering
+
+prop_compareAntiSymmetric
+    :: (SeqState DummyTarget, ShowFmt Address, ShowFmt Address)
+    -> Property
+prop_compareAntiSymmetric (s, ShowFmt a1, ShowFmt a2) =
+    fst (isOurs a1 s) && fst (isOurs a2 s) ==> cover 90 (a1 /= a2) "a1 /= a2" prop
```

Comment:
I wonder what is distribution to pick two `isOurs` wallets (hopefully precondition won't take too long - depending on the generator)

Reply:
```
      1) Cardano.Wallet.Primitive.AddressDiscovery.CompareDiscovery compareDiscovery is anti-symmetric
           *** Gave up! Passed only 199 tests; 232 discarded tests (98.5% a1 /= a2).
```
almost :P

Reply:
Ho damn :o ! The coverage is too tight maybe here ^^"